src/Microsoft.Health.Dicom.Functions/Update/Models/CleanupBlobArguments.cs

```diff
@@ -0,0 +1,22 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Microsoft.Health.Dicom.Core.Features.Model;
+using Microsoft.Health.Dicom.Core.Features.Partitioning;
+
+namespace Microsoft.Health.Dicom.Functions.Update.Models;
+
+public sealed class CleanupBlobArguments
+{
+    public IReadOnlyList<InstanceFileState> InstanceWatermarks { get; }
+    public Partition Partition { get; }
+
+    public CleanupBlobArguments(IReadOnlyList<InstanceFileState> instanceWatermarks, Partition partition)
```

Comment:
nit: I think you can skip the ctor if you're not going to assert anything and add initializers to the properties via the `init` keyword